src/workspaces/SettingsModal/SettingsModal.tsx

```diff
@@ -0,0 +1,176 @@
+      isBucketLifecycleSetting(setting)
+    ) as BucketLifecycleSetting[];
+    if (bucketLifecycleSettings.length > 0) {
+      if (printMultipleWarning && bucketLifecycleSettings.length > 1) {
+        // eslint-disable-next-line no-console
+        console.log('Multiple bucket lifecycle settings found, displaying only first');
+      }
+      return bucketLifecycleSettings[0];
+    }
+    return undefined;
+  };
+
+  const getDeleteLifecycleRule = (
+    bucketLifecycleSetting: BucketLifecycleSetting | undefined,
+    printMultipleWarning = false
+  ): DeleteBucketLifecycleRule | undefined => {
+    const configRules = bucketLifecycleSetting?.config.rules;
+    const deleteRules: DeleteBucketLifecycleRule[] = configRules?.filter((rule) =>
+      isDeleteBucketLifecycleRule(rule)
+    ) as DeleteBucketLifecycleRule[];
+    if (printMultipleWarning && !!deleteRules && deleteRules.length > 1) {
+      // eslint-disable-next-line no-console
+      console.log('Multiple delete bucket lifecycle rules found, displaying only first');
+    }
+    return !!deleteRules && deleteRules.length >= 1 ? deleteRules[0] : undefined;
+  };
+
+  useEffect(() => {
+    const loadSettings = withErrorReporting('Error loading workspace settings')(async () => {
+      const settings = (await Ajax(signal)
+        .Workspaces.workspaceV2(namespace, name)
+        .getSettings()) satisfies WorkspaceSetting[];
+      setWorkspaceSettings(settings);
+      const bucketLifecycleSetting = getFirstBucketLifecycleSetting(settings, true);
+      if (bucketLifecycleSetting !== undefined) {
+        const rule = getDeleteLifecycleRule(bucketLifecycleSetting, true);
+        if (rule) {
+          setLifecycleRulesEnabled(true);
+          const prefixes = rule.conditions.matchesPrefix;
+          if (prefixes.length === 0) {
+            setPrefixes([suggestedPrefixes.allObjects]);
+          } else {
+            setPrefixes(prefixes);
+          }
+          setLifecycleAge(rule.conditions.age);
+        }
+      }
+    });
+
+    loadSettings();
+  }, [namespace, name, signal]);
+
+  const persistSettings = withErrorReporting('Error saving workspace settings')(async () => {
+    let newSettings: WorkspaceSetting[];
+    if (lifecycleRulesEnabled) {
+      const prefixesOrNone = _.without([suggestedPrefixes.allObjects], prefixes);
+      newSettings = modifyFirstBucketDeletionRule(workspaceSettings || [], lifecycleAge!, prefixesOrNone);
+    } else {
+      newSettings = removeFirstBucketDeletionRule(workspaceSettings || []);
+    }
+    await Ajax().Workspaces.workspaceV2(namespace, name).updateSettings(newSettings);
+    props.onDismiss();
+    const originalLifecycleSetting = getFirstBucketLifecycleSetting(workspaceSettings || []);
+    const newLifecycleSetting = getFirstBucketLifecycleSetting(newSettings);
+    if (!_.isEqual(originalLifecycleSetting, newLifecycleSetting)) {
```

Comment:
@admull the event for changing bucket lifecycle settings will contain look like this:

<img width="643" alt="image" src="https://github.com/user-attachments/assets/52689a5c-aba7-4eb8-af11-a58d420bc9b2">

"prefix" will be one of the following 5 options: "Custom", "AllObjects", "AllSubmissions", "SubmissionsIntermediaries", or "AllSubmissionsAndSubmissionsIntermediaries".